### PR TITLE
chore: Merge upstream changes as of 07-08-2021

### PR DESCRIPTION
**Pull request by nearora-msft**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: deploy/csi-azuredisk-node-windows.yaml [hunk 1]
```diff
@@ -94,9 +94,6 @@ spec:
             - containerPort: 29603
               name: healthz
               protocol: TCP
-            - containerPort: 29605
```
Comment (mukhoakash) [inline deploy/csi-azuredisk-node-windows.yaml:97]: Do we need this change in deploy/v2,0.0.-alpha1.1 directory?

File: deploy/csi-azuredisk-node.yaml [hunk 2]
```diff
@@ -88,9 +88,6 @@ spec:
             - containerPort: 29603
               name: healthz
               protocol: TCP
-            - containerPort: 29605
```
Comment (mukhoakash) [inline deploy/csi-azuredisk-node.yaml:91]: Same regarding the deploy/v2.0.0-alpha.1 folder?

File: pkg/azuredisk/azuredisk_v2.go [hunk 3]
```diff
@@ -132,6 +132,7 @@ func newDriverV2(nodeID string,
 	driver.Version = driverVersion
 	driver.NodeID = nodeID
 	driver.VolumeAttachLimit = volumeAttachLimit
+	driver.volumeLocks = volumehelper.NewVolumeLocks()
```
Comment (mukhoakash) [inline pkg/azuredisk/azuredisk_v2.go:135]: Already present in line 137

File: pkg/azuredisk/nodeserver_test.go [hunk 4]
```diff
@@ -718,14 +718,14 @@ func TestNodeExpandVolume(t *testing.T) {
 
 	invalidPathErr := testutil.TestError{
 		DefaultError: status.Error(codes.NotFound, "failed to determine device path for volumePath [./test]: path \"./test\" does not exist"),
-		WindowsError: status.Errorf(codes.NotFound, "Forward to GetVolumeIDFromTargetPath failed, err=error getting the volume for the mount ./test, internal error error getting volume from mount. cmd: (Get-Item -Path ./test).Target, output: (Get-Item -Path ), error: path \"./test\" does not exist"),
```
Comment (mukhoakash) [inline pkg/azuredisk/nodeserver_test.go:721]: Please dont revert these changes. They were breaking the windows unit test cases in the last merge.
Same goes for the below changes as well.

File: pkg/azuredisk/nodeserver_test.go [hunk 5]
```diff
@@ -718,14 +718,14 @@ func TestNodeExpandVolume(t *testing.T) {
...
+		WindowsError: status.Errorf(codes.NotFound, "Forward to GetVolumeIDFromTargetPath failed, err=error getting the volume for the mount .\\test, internal error error getting volume from mount. cmd: (Get-Item -Path .\\test).Target, output: , error: <nil>"),
```
Comment (mukhoakash) [inline pkg/azuredisk/nodeserver_test.go:721]: I see that you have fixed the fakes mounter for windows. Be aware of the `/` and `\` as they tend to create mismatch and failures.

File: pkg/azuredisk/nodeserver_test.go (hunk 4 above)
Comment (nearora-msft) [reply pkg/azuredisk/nodeserver_test.go:721]: Which changes should I not revert?

File: deploy/csi-azuredisk-node.yaml (hunk 2 above)
Comment (nearora-msft) [reply deploy/csi-azuredisk-node.yaml:91]: Added

File: pkg/azuredisk/azuredisk_v2.go (hunk 3 above)
Comment (nearora-msft) [reply pkg/azuredisk/azuredisk_v2.go]: Removed the duplicate one.

File: pkg/azuredisk/nodeserver_test.go (hunk 4 above)
Comment (nearora-msft) [reply pkg/azuredisk/nodeserver_test.go:721]: Keeping these changes and made some changes to handle the path failures in fake_safe_mounter_windows.go

**Review by mukhoakash — APPROVED** [review]

**Comment by mukhoakash** [comment]
As discussed with @nearora-msft , we will track the e2e error in a separate PR since we have established that this is not related to this PR.
